python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1775 @@
+        else:
+            donut_data["fx"] = [0.0] * num_donuts
+            donut_data["fy"] = [0.0] * num_donuts
+
+        # Get SNR from TARTS
+        if hasattr(self.tarts, 'SNR'):
+            snr_list = (
+                self.tarts.SNR.cpu().numpy().tolist()
+                if hasattr(self.tarts.SNR, 'cpu')
+                else list(self.tarts.SNR)
+            )
+
+            # Flatten nested lists if needed
+            if snr_list and isinstance(snr_list[0], list):
+                snr_list = [item for sublist in snr_list for item in sublist]
+
+            if len(snr_list) == num_donuts:
+                donut_data["snr"] = snr_list
+            else:
+                donut_data["snr"] = [snr_list[0] if len(snr_list) > 0 else np.nan] * num_donuts
+        else:
+            donut_data["snr"] = [np.nan] * num_donuts
+
+        # Add detector information as a column (required by downstream
+        # aggregation)
+        # Ensure detector name is consistent with expected format
+        donut_data["detector"] = [detector_name] * num_donuts
+        self.log.info("Using detector name: '%s' for %d donuts (defocal type: '%s')",
+                     detector_name, num_donuts, defocalType)
+
+        # Add missing telescope coordinate columns required by downstream
+        # aggregation
+        # These are not generated by neural network, so use default values
+        donut_data["thx_CCS"] = [np.nan] * num_donuts  # Telescope X in Camera Coordinate System
+        donut_data["thy_CCS"] = [np.nan] * num_donuts  # Telescope Y in Camera Coordinate System
+        donut_data["thx_OCS"] = [np.nan] * num_donuts  # Telescope X in Observatory Coordinate System
+        donut_data["thy_OCS"] = [np.nan] * num_donuts  # Telescope Y in Observatory Coordinate System
+        donut_data["th_N"] = [np.nan] * num_donuts      # Telescope North coordinate
+        donut_data["th_W"] = [np.nan] * num_donuts      # Telescope West coordinate
+
+        # Create the table
+        donut_table = QTable(donut_data)
+
+        # Add units matching standard format
+        donut_table["coord_ra"].unit = u.rad  # Use radians like standard
+        donut_table["coord_dec"].unit = u.rad  # Use radians like standard
+        # Note: centroid_x and centroid_y are kept as plain floats to avoid
+        # Point2D constructor issues
+        donut_table["fx"].unit = u.deg
+        donut_table["fy"].unit = u.deg
+
+        # Add comprehensive visit_info metadata matching standard format
+        try:
+            visitInfo = exposure.visitInfo
+
+            # Get visit ID
+            try:
+                exp_id = exposure.getInfo().getId()
+                if hasattr(exp_id, 'getVisitId'):
+                    visit_id = exp_id.getVisitId()
+                else:
+                    visit_id = exp_id
+            except (AttributeError, NameError):
+                visit_id = visitInfo.id
```

Comment:
When/why is this missing? Generally you should avoid so many `try/excepts` and also have warnings if you're inside them so you don't get silently different behaviour.